Indicators/NormalizedAverageTrueRange.cs

```diff
@@ -47,17 +47,19 @@ public class NormalizedAverageTrueRange : BarIndicator
         /// </summary> 
         /// <param name="period">The period of the NATR</param>
         public NormalizedAverageTrueRange(int period)
-            : this("NATR" + period, period)
+            : this($"NATR({period})", period)
         {
         }
 
         /// <summary>
         /// Gets a flag indicating when this indicator is ready and fully initialized
         /// </summary>
-        public override bool IsReady
-        {
-            get { return Samples > _period; }
-        }
+        public override bool IsReady => Samples > _period;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod => _period + 1;
```

Comment:
I didn't think of this on the previous PR, but I think these definitions might be better phrased against the dependency, for example, here we could say `=> _smoother.WarmUpPeriod` -- and going even further, could make `SimpleMovingAverage.WarmUpPeriod => RollingSum.WarmUpPeriod` (which would require `RollingWindow<T> : IWarmUpPeriodProvider`

---

#### History of +1 warm up periods

My reasoning here comes down to this infamous `+ 1` that often appears and IIRC, it was a _very_ [early](b3de52091) decision made due to `RollingWindow<T>.MostRecentlyRemoved` throwing an exception if we haven't removed any items yet. Thinking back on that decision now it seems silly and could potentially lead to variations in results vs different platforms, especially in the indicators that use a different formula for the warmup period (such as, technically, wilders).

I'm not recommending a change here, but simply wanted to share some knowledge on why this is like this (SMA10 requires 11 points to be `IsReady==true`).

@jaredbroad @StefanoRaggi @Martin-Molinero - tagged for knowledge transfer :) feel free to ping me to discuss further

Reply:
When we have an indicator that inherits from `WindowIndicator<IndicatorDataPoint>`, like SMA, we get the following `WarmUpPeriod`:
```csharp
public int WarmUpPeriod => Period;
```
So, we would have:
```csharp
public class RollingWindow<T> : IReadOnlyWindow<T>, IIndicatorWarmUpPeriodProvider
{
    public int WarmUpPeriod => Size;
}
```
and
```csharp
public abstract class WindowIndicator<T> : IndicatorBase<T>, IIndicatorWarmUpPeriodProvider
{
    public int WarmUpPeriod => Period;   // since public int Period => _window.Size;
}
```

However, this particular case is different, because IsReady is defined as:
```csharp
public override bool IsReady => Samples > _period;
```
where `IsReady` in `WindowIndicator`/`RollingWindow` is a greater **or equal** condition.